packages/core/src/Prelude/Filterable/index.ts

```diff
@@ -0,0 +1,75 @@
+import type { Either } from "../../Classic/Either"
+import type { Option } from "../../Classic/Option"
+import type { Predicate, Refinement } from "../../Function"
+import type { Separated } from "../../Utils"
+import type { Compactable } from "../Compactable"
+import type { Covariant } from "../Covariant"
+import type * as HKT from "../HKT"
+
+export interface Filterable<F extends HKT.URIS, C = HKT.Auto>
+  extends Covariant<F, C>,
+    Compactable<F, C> {
+  readonly filter: Filter<F, C>
+  readonly partition: Partition<F, C>
+  readonly filterMap: FilterMap<F, C>
+  readonly partitionMap: PartitionMap<F, C>
+}
+
+export interface Filter<F extends HKT.URIS, C = HKT.Auto> {
+  <A, B extends A>(refinement: Refinement<A, B>): <
+    N extends string,
+    K,
+    Q,
+    W,
+    X,
+    I,
+    S,
+    R,
+    E
+  >(
+    fa: HKT.Kind<F, C, N, K, Q, W, X, I, S, R, E, A>
+  ) => HKT.Kind<F, C, string, K, Q, W, X, I, S, R, E, B>
```

Comment:
Shouldn't this `string` be an `N` type?

Reply:
Arguable, I do think N would be a better type here too but the concept of filter is to remove stuff, meaning this N would actually be oversized